crates/ide-diagnostics/src/handlers/json_is_not_rust.rs

```diff
@@ -0,0 +1,201 @@
+//! This diagnostic provides an assist for creating a struct definition from a json
+//! example.
+
+use ide_db::{base_db::FileId, source_change::SourceChange};
+use itertools::Itertools;
+use stdx::format_to;
+use syntax::{
+    ast::{self, make},
+    SyntaxKind, SyntaxNode,
+};
+use text_edit::TextEdit;
+
+use crate::{fix, Diagnostic, Severity};
+
+#[derive(Default)]
+struct State {
+    result: String,
+    struct_counts: usize,
+}
+
+impl State {
+    fn generate_new_name(&mut self) -> ast::Name {
+        self.struct_counts += 1;
+        make::name(&format!("Struct{}", self.struct_counts))
+    }
+
+    fn build_struct(&mut self, value: &serde_json::Map<String, serde_json::Value>) -> ast::Type {
+        let name = self.generate_new_name();
+        let ty = make::ty(&name.to_string());
+        let strukt = make::struct_(
+            None,
+            name,
+            None,
+            make::record_field_list(value.iter().sorted_unstable_by_key(|x| x.0).map(
+                |(name, value)| make::record_field(None, make::name(name), self.type_of(value)),
+            ))
+            .into(),
+        );
+        format_to!(self.result, "#[derive(Serialize, Deserialize)]\n{}\n", strukt);
```

Comment:
It would be nice to search for whether the `Serialize` and `Deserialize` attributes exist in the depdenncies somewhere and only emit these if they do + inserting imports if required

Reply:
I did something here, but it seems it doesn't respect import config and always add separated `::serde::Deserialize` and `::serde::Serialize`. What is wrong?

Reply:
You'll need to resolve the `::serde::Deserialize` path to the attribute first, then query for an import path for the resolved definition and insert that. (Note resolving that path might yield the trait instead, since our path resolution in the IDE layer currently ignores the fact that a path resolves in up to 3 namespaces)

Reply:
Now it works, but I didn't find out how to force resolve to be attribute and not trait. I'm currently using `speculative_resolve` method.

Reply:
Ye, you would need to add an extra function that resolves only in macro namespace here for the time being. Though I don't think its too big of a problem if we find the trait instead of the macro here right now.